### ISSUE TITLE
a11y: Fix paths a11y inputs for Fleet Git Repo

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#13831]
**Setup**
- Rancher version:
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

**a11y issue 198**
_Missing visual instruction for form fields_

The mentioned instruction is not visually available, it is available as placeholder text within input fields. However, it gets disappear while typing resulting in cognitive and screen reader users might not be able to identify the mandatory fields.

Paths section
- e.g. /directory/in/your/repo

**a11y issue 199**
Identical label for buttons

Identical labels are specified for the buttons that resulted in screen reader users finding it difficult to understand the purpose of the buttons effectively.

Git Repo - Add Repository
- Remove buttons (available on activating "Add Path" button)

Repository: Create
- Remove buttons (available on activating "Add Label" and "Add Annotation" buttons)

Home Links
- Remove buttons (available on activating "Add Link" button) 

The screen reader should announce unique and descriptive label for the buttons.

Specify unique and descriptive labels such as "Remove 1", "Remove 2", "Remove 3" and so on for the mentioned buttons using "aria-label" attribute.

**To Reproduce**

**Result**

**Expected Result**

**Screenshots**

**Additional context**

**Comment by izaac** [comment]
Rancher version: `v2.12-e9f51051f414b4103fae57c8e336866e9761cee1-head`
Rancher UI version: `master ff12e4a`


Paths, labels and annotations rows are described properly by their aria-label object and number. In both Edit and Create workflows.
